### DISCUSSION TITLE
Can we cache particular dashboard and refresh according to our given time interval?

### DISCUSSION THREAD

**Discussion by Ashish7129** [apache/superset#18328] [Q&A / Help]
Hi all,

I am using superset for the last two months. I must say It is a beautiful tool for analytics. I came across one good problem in which I have to cache a particular dashboard and will refresh in a particularly given time frame. If User A, go to the dashboard (D1), which contains different charts with data. It must take some time to load all the data on the charts. If User B, go to the same dashboard (D1), it should see the charts faster as it was cached earlier by User A.  I want the whole dashboard would be kept in cache for some time and would be refresh in the given time frame for that dashboard. I am currently on superset 1.2.0 version. Can someone help me to find a prominent solution for this? 

Can we cache the whole dashboard? It should be viewed faster for another user for a given timeframe. Can we refresh the particular dashboard with the given time frame? 

Or Can we add the different caching time for different dashboards or we can say can we specify for each dashboard uniquely ?

**Comment by mayurnewase** [comment]
All charts in dashboard will be cached independently and you can set default timeout in cache setting so after that charts will get new data.

**Comment by Ashish7129** [comment]
@mayurnewase, I want a different dashboard that will refresh according to a different timeout. I don't want to use the refresh dashboard button (which is for a particular session). Can I do this?

**Comment by Ashish7129** [comment]
@amitmiran137 Any suggestions ?

**Comment by Ashish7129** [comment]
@mayurnewase, can you please help me in the steps that how can we achieve this? I tried and add seconds in the chart settings but it didn't work for me. It should have automatically pulled the new data but it didn't.

**Comment by Ashish7129** [comment]
@mayurnewase Thanks for your reply, now I understood but I have one more query related to the calling of the new data for the multiple users, Suppose if we set the cache timeout into the charts and have 10-20 users come after the cache timeout at a given point of time. All users who open the same dashboard or charts, so will it be the single call of new data and the rest of the cache data or will it be the call of new data for every user?

**Comment by nvlong198** [comment]
![image](https://user-images.githubusercontent.com/23328191/184793117-fd1c015d-0c03-4b14-9133-7951e3f8db05.png)
**I hope it will have a schedule refresh dashboard alongside refresh refresh interval.**

**Reply by nvlong198** [reply]
@mayurnewase could you guys consider my requests? 
the problem is I want to preload the dashboard before peak time using it, tables with a large number of records and complex queries can not be the short-time response.

**Comment by mtrentz** [comment]
Any updates on this that I might be missing? I'm looking for a way to refresh charts with a set interval so that when it's peak use the queries will be cached.

**Reply by conmurray** [reply]
+1 to this. I came here looking for the exact same feature

**Comment by aravind-naidu** [comment]
+1 Don't want to hit Refresh Dashboard every time I need to refresh. Some of my dashboards need to be refreshed every hour, and some at 24 hour interval. 

Anybody has solved this yet?
